plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionContext.kt

```diff
@@ -41,3 +45,40 @@ interface StmtConversionContext<out RTC : ResultTrackingContext> : MethodConvers
         return ctx.resultCtx.resultVar
     }
 }
+
+fun <RTC : ResultTrackingContext> StmtConversionContext<RTC>.embedPropertyAccess(symbol: FirPropertyAccessExpression): PropertyAccessEmbedding =
+    when (val calleeSymbol = symbol.calleeSymbol) {
+        is FirValueParameterSymbol -> LocalPropertyAccess(embedValueParameter(calleeSymbol))
+        is FirPropertySymbol ->
+            when (val receiverFir = symbol.dispatchReceiver) {
+                null -> LocalPropertyAccess(embedLocalProperty(calleeSymbol))
+                else -> ClassPropertyAccess(convert(receiverFir), embedGetter(calleeSymbol), embedSetter(calleeSymbol))
+            }
+        else -> throw Exception("Property access symbol $calleeSymbol has unsupported type.")
+    }
+
+@OptIn(SymbolInternals::class)
+fun <RTC : ResultTrackingContext> StmtConversionContext<RTC>.embedGetter(symbol: FirPropertySymbol): GetterEmbedding? =
+    when (val getter = symbol.fir.getter) {
+        null -> null
+        is FirDefaultPropertyGetter -> BackingFieldGetter(
+            VariableEmbedding(
+                symbol.callableId.embedName(),
+                embedType(symbol.resolvedReturnType)
+            )
+        )
+        else -> CustomGetter(embedFunction(getter.symbol))
+    }
+
+@OptIn(SymbolInternals::class)
+fun <RTC : ResultTrackingContext> StmtConversionContext<RTC>.embedSetter(symbol: FirPropertySymbol): SetterEmbedding? =
+    when (val setter = symbol.fir.setter) {
+        null -> null
+        is FirDefaultPropertySetter -> BackingFieldSetter(
+            VariableEmbedding(
+                symbol.callableId.embedName(),
+                embedType(symbol.resolvedReturnType)
+            )
+        )
+        else -> CustomSetter(embedFunction(setter.symbol))
+    }
```

Comment:
I don't understand why do you need to define these functions outside of the class

Reply:
It's not strictly necessary, but note that if they are in the context then they are functions that can be overriden, which doesn't make sense here: we only want one instance of these functions, overriding them would be very strange.

Reply:
makes sense